web/index.php

```diff
@@ -33,117 +31,145 @@
 $app->container->singleton('opentok', function () {
-        return new OpenTok(getenv('API_KEY'), getenv('API_SECRET'));
+    return new OpenTok(getenv('TOKBOX_API_KEY'), getenv('TOKBOX_SECRET'));
 });
 
 // Store the API Key in the app container
-$app->apiKey = getenv('API_KEY');
+$app->apiKey = getenv('TOKBOX_API_KEY');
 
-// If a sessionId has already been created, retrieve it from the storage
-$app->container->singleton('sessionId', function() use ($app) {
-    if ($app->storage->exists('sessionId')) {
-        return $app->storage->retrieve('sessionId');
-    }
+$app->get('/', cors, function () use ($app) {
+    $app->render('home.php');
+});
 
-    $session = $app->opentok->createSession(array(
-        'mediaMode' => MediaMode::ROUTED
-    ));
-    $app->storage->store('sessionId', $session->getSessionId());
-    return $session->getSessionId();
+/**
+ * GET /session redirects to /room/session
+ */
+$app->get('/session', 'cors', function () use ($app) { 
+    $app->redirect('/room/session');
 });
 
-// Route to return the SessionID and token as a json
-$app->get('/session', 'cors', function () use ($app) {
+/**
+ * GET /room/:name
+ */
+$app->get('/room/:name', 'cors', function($name) use ($app) {
 
-    $token = $app->opentok->generateToken($app->sessionId);
+    // if a room name is already associated with a session ID
+    if ($app->storage->exists($name)) {
 
-    $responseData = array(
-        'apiKey' => $app->apiKey,
-        'sessionId' => $app->sessionId,
-        'token'=>$token
-    );
+        // fetch the sessionId from local storage
+        $app->sessionId = $app->storage[$name];
 
-    $app->response->headers->set('Content-Type', 'application/json');
-    echo json_encode($responseData);
+        // generate token
+        $token = $app->opentok->generateToken($app->sessionId);
+        $responseData = array(
+            'apiKey' => $app->apiKey,
+            'sessionId' => $app->sessionId,
+            'token'=>$token
+        );
+
+        $app->response->headers->set('Content-Type', 'application/json');
+        echo json_encode($responseData);
+    }
+    else {
+        $session = $app->opentok->createSession(array(
+            'mediaMode' => MediaMode::ROUTED
+        ));
+
+        // store the sessionId into local
+        $app->storage[$name] = $session->getSessionId();
+        
+        // generate token
+        $token = $app->opentok->generateToken($session->getSessionId());
+        $responseData = array(
+            'apiKey' => $app->apiKey,
+            'sessionId' => $session->getSessionId(),
+            'token'=>$token
+        );
+
+        $app->response->headers->set('Content-Type', 'application/json');
+        echo json_encode($responseData);
+    }
 });
 
-// Start Archiving and return the Archive ID
-$app->post('/start/:sessionId', 'cors', function ($sessionId) use ($app) {
+/**
+ * POST /archive/start
+ */
+$app->post('/archive/start', 'cors', function () use ($app) {
```

Comment:
According to https://tokbox.atlassian.net//browse/ECO-213, the route should be /session/:sessionId/archive/start.

Was this changed?